src/convertToObject.js

```diff
@@ -1,12 +1,42 @@
 'use strict';
 
+/**
+ * @param {string} sourceString
+ *
+ * @return {object}
+ */
+'use strict';
+
 /**
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  if (!sourceString.trim()) {
+    return {};
+  }
+
+  const rules = sourceString.split(';').filter(Boolean);
+
+  const result = {};
+
+  for (const rule of rules) {
+    const [key, ...valueParts] = rule.split(':');
```

Comment:
Consider handling cases where the key or value might contain colons. The current implementation splits on the first colon, which may not be the intended behavior if colons are part of the value.